### DISCUSSION TITLE
Search filter-lists by content

### DISCUSSION THREAD

**Discussion by Rudxain** [uBlockOrigin/uBlock-issues#3367] [Ideas]
As of v1.59, the search-box in the "Filter lists" tab can only search filter list by name ([erratum](https://github.com/uBlockOrigin/uBlock-issues/discussions/3367#discussioncomment-10758716)).

I propose 2 options:
- Remove the search-box: This will make development easier for maintainers, and will free up screen and storage space.
- Allow searching the **contents** of filter-lists, as if the asset viewer was open for every file (is this "recursive"?)

For content search, it doesn't have to be "advanced" with highlighting or anything, just show the lists that contain the sub-string

**Comment by garry-ut99** [comment]
_As for the option 1_: 
- the uBO's search box in "Filter lists" (which has been implemented a year ago, in 06.2023 in uBO 1.50.0) is too handy to be removed, it's much useful in a comparison to a browser's search box, as it was specifically designed for uBO, while browser's search box wasn't

_As for the option 2:_ a duplicate of:
- https://github.com/uBlockOrigin/uBlock-issues/issues/205

**Reply by Rudxain** [reply]
The only use-case I can think of is taking screenshots of related lists, to document which ones are enabled on a user's browser. Is there any other use-case for which it's orders-of-magnitude better than a browser's "search in page"?

And thank you for the issue link! I tried to search for dupes, but there were ~200 results

**Reply by garry-ut99** [reply]
> The only use-case I can think of is taking screenshots of related lists, to document which ones are enabled on a user's browser. 

In such cases screenshoots are not so much necessary, as it was already solved by implementing in uBO a "Troubleshooting" tab, where at the bottom, there are all required informations displayed, including filter lists, and they are then embeded automatically in issue reports on uAssets in a spoiler, maybe only sometimes when it shows "too many", screenshoots will be useful.

> Is there any other use-case for which it's orders-of-magnitude better than a browser's "search in page"?

Yes:
- [1]: when searching for filter list names via uBO search box, only filter lists which contain the text are displayed, and all other filters lists are hidden, this helps much to find lists faster and easier
- [2] when searching for filter list names via uBO search box, it searchs also filter lists which are collapsed and invisible, this helps much to find lists faster and easier, as an user doesn't need to uncollapse them before searching (which would require to click on Network and Cosmetic filters counter at the top)
- [3] uBO search box supports an incremental search:
   - Firefox and some other browsers don't support it, 
   and extensions like [{find+}](https://addons.mozilla.org/pl/firefox/addon/brandon1024-find/) don't work in uBlock because they don't work in other extensions pages
   - Chromium browsers support it too, but its slower

<hr>

> I tried to search for dupes, but there were ~200 results

Then your search method is wrong, you should always first limit your search to titles only, in the current case it gave me only 16 results incudling the original issue (of which your issue is a duplicate of):
- https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue+in%3Atitle+search

**Reply by Rudxain** [reply]
> already solved by implementing in uBO a "Troubleshooting" tab

I agree. For that use-case, the tab is much better.

> [1]

IMO: +1% convenience. +8% "noise" reduction.
 
> [2]

IMO: +2% convenience.

> [3]

+8% convenience. ~ -5% performance? (unsure)

Final (personal) score: +11% convenience, +8% focus, ~ -5% perf. Not bad.

I believe other users will find it more useful, so the feature should stay. You've convinced me!

> limit your search to titles only

I'm sometimes afraid of doing that, then not finding what I'm looking for. But the strategy of doing a narrow-search followed by a broad-search might be helpful sometimes. Thanks for the suggestion 👍

**Reply by garry-ut99** [reply]
> Final (personal) score: +11% convenience, +8% focus, ~ -5% perf. Not bad.
> I believe other users will find it more useful, so the feature should stay. You've convinced me!

I'm not sure but I guess that was supposed to be sarcasm, if so, then you've denied my points, providing some `%` based calculations, which noone have clue about how they were calculated (except you), so I'm unable to reply to it.

<hr>

> >  [3]
> 
> +8% convenience. ~ -5% performance? (unsure)

There is a about 0,5-1 second delay after searching using Chromium search box before results appear, which might be annoying: [Chromium video](https://github.com/user-attachments/assets/efbf4f2c-c130-49bd-b8a1-4c37fea6cb6b). In uBO search box, results are instant after typing every letter (I guess perhaps 50ms - 100ms): [uBO video](https://github.com/user-attachments/assets/201b58fb-df43-46b6-9f50-83a8cafc4b9d)

**Reply by Rudxain** [reply]
> sarcasm

I **genuinely** believe the feature is useful for _some_ users, but (for me) it's not so helpful :/ . I don't even use the browser's built-in search often, as the "Filter lists" tab isn't long, and I rarely visit it. I'm mostly concerned about my Dyn-Rules, where the search-box is **very** helpful, as it searches the contents of a "big" list.

> providing some `%` based calculations, which noone have clue about how they were calculated (except you)

I'm not even sure how I calculated either 😅. Roughly: 0% = "absolutely no diff", +25% = "noticeably better than no search-box", +50% = "considerably better than no search-box", +100% = "2x better than no search-box" (1 binary order of magnitude). But even those terms are vague and subjective, so I'm not sure how to properly measure or communicate them 😕

> There is a about 0,5-1 second delay after searching using Chromium search box before results appear, which might be annoying: [Chromium video](https://github.com/user-attachments/assets/efbf4f2c-c130-49bd-b8a1-4c37fea6cb6b). In uBO search box, results are instant after typing every letter (I guess perhaps 50ms-100ms): [uBO video](https://github.com/user-attachments/assets/201b58fb-df43-46b6-9f50-83a8cafc4b9d)

I meant "perf of not searching anything" (always negative for any search implementation). But it makes more sense to measure WRT the browser search, as it's more fair. In this case it's between ~10% and ~0%

**Reply by garry-ut99** [reply]
> I don't even use the browser's built-in search often, as the "Filter lists" tab isn't long

Depends, on Desktop on Default when all lists are collapsed, they have vertically length of 1 - 1.5 screen which isn't much (on Mobile it's 2 screens), but when all lists are uncollapsed they have vertically length of 5 screens (5.5 with some user's custom filter lists imported) which I think is considered long enought to have a dedicated search box implemented.

<hr>

Given that your removal request cannot be met because the current discussion has proved that uBO's search box was worth implementing (for users who search filter lists), lets assume your request has been automatically transformed/converted/downgraded to: "add possibility to toggle on/off uBO's search box in Filter lists" (and in case that it would ever happen, I think it should be still enabled by default, otherwise some users could miss it and never know about it).

**Reply by Rudxain** [reply]
The toggle seems like a nice middle-ground. I like that idea

**Comment by Rudxain** [comment]
I've searched "annoyance", and the results show:
- "Cookie notices"
- "Social widgets"
- "Annoyances"

This obviously implies the search-box does more than just match titles
